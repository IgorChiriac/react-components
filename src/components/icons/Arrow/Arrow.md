This component is a wrapper around the svg code for Arrow's icon.

Allowed directions are 'down' | 'up' | 'left' | 'right' | number | string

```jsx
<Arrow color="blue" direction="down"/>
<Arrow color="blue" direction="up"/>
<Arrow color="blue" direction="left"/>
<Arrow color="blue" direction="right"/>
<Arrow color="blue" direction={45}/>
<Arrow color="blue" direction="225"/>
```
