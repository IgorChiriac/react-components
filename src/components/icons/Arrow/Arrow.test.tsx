import React from 'react';
import { render } from 'react-testing-library';
import Arrow from './Arrow';

describe('<Arrow />', () => {
  const directions = ['down', 'up', 'left', 'right', 45, '225'];
  directions.forEach(direction => {
    it(`renders the component pointing ${direction}`, () => {
      const { container } = render(<Arrow color="black" direction={direction} />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
