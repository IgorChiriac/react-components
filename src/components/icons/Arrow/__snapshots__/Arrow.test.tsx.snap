// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Arrow /> renders the component pointing 45 1`] = `
.c0 {
  -webkit-transition: -webkit-transform 0.2s;
  -webkit-transition: transform 0.2s;
  transition: transform 0.2s;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

<svg
  aria-label="Arrow pointing 45"
  class="c0"
  direction="45"
  height="12px"
  role="img"
  viewBox="0 0 11 5"
  width="12px"
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    arrow pointing 
    45
  </title>
  <g
    fill="none"
    fill-rule="evenodd"
    id="Symbols"
    stroke="none"
    stroke-width="1"
  >
    <g
      fill="black"
      fill-rule="nonzero"
      id="accordion/open"
      transform="translate(0.000000, -14.000000)"
    >
      <g
        id="Group"
      >
        <path
          d="M7.63149316,17.3813356 L4.28255792,21.7152282 C3.98915505,22.0949239 3.51345502,22.0949239 3.22005215,21.7152282 C3.07915512,21.5328919 3,21.2855908 3,21.0277282 L3,11.9722718 C3,11.4353009 3.33637072,11 3.75130504,11 C3.95056352,11 4.1416609,11.1024355 4.28255792,11.2847718 L7.63149316,15.6186645 C7.86830411,15.925124 7.92952055,16.0681625 7.96979452,16.2401176 C8.01006849,16.4120727 8.01006849,16.5879273 7.96979452,16.7598824 C7.92952055,16.9318375 7.86830411,17.074876 7.63149316,17.3813356 Z"
          id="path-1"
          transform="translate(5.500000, 16.500000) rotate(90.000000) translate(-5.500000, -16.500000) "
        />
      </g>
    </g>
  </g>
</svg>
`;

exports[`<Arrow /> renders the component pointing 225 1`] = `
.c0 {
  -webkit-transition: -webkit-transform 0.2s;
  -webkit-transition: transform 0.2s;
  transition: transform 0.2s;
  -webkit-transform: rotate(225deg);
  -ms-transform: rotate(225deg);
  transform: rotate(225deg);
}

<svg
  aria-label="Arrow pointing 225"
  class="c0"
  direction="225"
  height="12px"
  role="img"
  viewBox="0 0 11 5"
  width="12px"
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    arrow pointing 
    225
  </title>
  <g
    fill="none"
    fill-rule="evenodd"
    id="Symbols"
    stroke="none"
    stroke-width="1"
  >
    <g
      fill="black"
      fill-rule="nonzero"
      id="accordion/open"
      transform="translate(0.000000, -14.000000)"
    >
      <g
        id="Group"
      >
        <path
          d="M7.63149316,17.3813356 L4.28255792,21.7152282 C3.98915505,22.0949239 3.51345502,22.0949239 3.22005215,21.7152282 C3.07915512,21.5328919 3,21.2855908 3,21.0277282 L3,11.9722718 C3,11.4353009 3.33637072,11 3.75130504,11 C3.95056352,11 4.1416609,11.1024355 4.28255792,11.2847718 L7.63149316,15.6186645 C7.86830411,15.925124 7.92952055,16.0681625 7.96979452,16.2401176 C8.01006849,16.4120727 8.01006849,16.5879273 7.96979452,16.7598824 C7.92952055,16.9318375 7.86830411,17.074876 7.63149316,17.3813356 Z"
          id="path-1"
          transform="translate(5.500000, 16.500000) rotate(90.000000) translate(-5.500000, -16.500000) "
        />
      </g>
    </g>
  </g>
</svg>
`;

exports[`<Arrow /> renders the component pointing down 1`] = `
.c0 {
  -webkit-transition: -webkit-transform 0.2s;
  -webkit-transition: transform 0.2s;
  transition: transform 0.2s;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

<svg
  aria-label="Arrow pointing down"
  class="c0"
  direction="down"
  height="12px"
  role="img"
  viewBox="0 0 11 5"
  width="12px"
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    arrow pointing 
    down
  </title>
  <g
    fill="none"
    fill-rule="evenodd"
    id="Symbols"
    stroke="none"
    stroke-width="1"
  >
    <g
      fill="black"
      fill-rule="nonzero"
      id="accordion/open"
      transform="translate(0.000000, -14.000000)"
    >
      <g
        id="Group"
      >
        <path
          d="M7.63149316,17.3813356 L4.28255792,21.7152282 C3.98915505,22.0949239 3.51345502,22.0949239 3.22005215,21.7152282 C3.07915512,21.5328919 3,21.2855908 3,21.0277282 L3,11.9722718 C3,11.4353009 3.33637072,11 3.75130504,11 C3.95056352,11 4.1416609,11.1024355 4.28255792,11.2847718 L7.63149316,15.6186645 C7.86830411,15.925124 7.92952055,16.0681625 7.96979452,16.2401176 C8.01006849,16.4120727 8.01006849,16.5879273 7.96979452,16.7598824 C7.92952055,16.9318375 7.86830411,17.074876 7.63149316,17.3813356 Z"
          id="path-1"
          transform="translate(5.500000, 16.500000) rotate(90.000000) translate(-5.500000, -16.500000) "
        />
      </g>
    </g>
  </g>
</svg>
`;

exports[`<Arrow /> renders the component pointing left 1`] = `
.c0 {
  -webkit-transition: -webkit-transform 0.2s;
  -webkit-transition: transform 0.2s;
  transition: transform 0.2s;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
}

<svg
  aria-label="Arrow pointing left"
  class="c0"
  direction="left"
  height="12px"
  role="img"
  viewBox="0 0 11 5"
  width="12px"
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    arrow pointing 
    left
  </title>
  <g
    fill="none"
    fill-rule="evenodd"
    id="Symbols"
    stroke="none"
    stroke-width="1"
  >
    <g
      fill="black"
      fill-rule="nonzero"
      id="accordion/open"
      transform="translate(0.000000, -14.000000)"
    >
      <g
        id="Group"
      >
        <path
          d="M7.63149316,17.3813356 L4.28255792,21.7152282 C3.98915505,22.0949239 3.51345502,22.0949239 3.22005215,21.7152282 C3.07915512,21.5328919 3,21.2855908 3,21.0277282 L3,11.9722718 C3,11.4353009 3.33637072,11 3.75130504,11 C3.95056352,11 4.1416609,11.1024355 4.28255792,11.2847718 L7.63149316,15.6186645 C7.86830411,15.925124 7.92952055,16.0681625 7.96979452,16.2401176 C8.01006849,16.4120727 8.01006849,16.5879273 7.96979452,16.7598824 C7.92952055,16.9318375 7.86830411,17.074876 7.63149316,17.3813356 Z"
          id="path-1"
          transform="translate(5.500000, 16.500000) rotate(90.000000) translate(-5.500000, -16.500000) "
        />
      </g>
    </g>
  </g>
</svg>
`;

exports[`<Arrow /> renders the component pointing right 1`] = `
.c0 {
  -webkit-transition: -webkit-transform 0.2s;
  -webkit-transition: transform 0.2s;
  transition: transform 0.2s;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
}

<svg
  aria-label="Arrow pointing right"
  class="c0"
  direction="right"
  height="12px"
  role="img"
  viewBox="0 0 11 5"
  width="12px"
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    arrow pointing 
    right
  </title>
  <g
    fill="none"
    fill-rule="evenodd"
    id="Symbols"
    stroke="none"
    stroke-width="1"
  >
    <g
      fill="black"
      fill-rule="nonzero"
      id="accordion/open"
      transform="translate(0.000000, -14.000000)"
    >
      <g
        id="Group"
      >
        <path
          d="M7.63149316,17.3813356 L4.28255792,21.7152282 C3.98915505,22.0949239 3.51345502,22.0949239 3.22005215,21.7152282 C3.07915512,21.5328919 3,21.2855908 3,21.0277282 L3,11.9722718 C3,11.4353009 3.33637072,11 3.75130504,11 C3.95056352,11 4.1416609,11.1024355 4.28255792,11.2847718 L7.63149316,15.6186645 C7.86830411,15.925124 7.92952055,16.0681625 7.96979452,16.2401176 C8.01006849,16.4120727 8.01006849,16.5879273 7.96979452,16.7598824 C7.92952055,16.9318375 7.86830411,17.074876 7.63149316,17.3813356 Z"
          id="path-1"
          transform="translate(5.500000, 16.500000) rotate(90.000000) translate(-5.500000, -16.500000) "
        />
      </g>
    </g>
  </g>
</svg>
`;

exports[`<Arrow /> renders the component pointing up 1`] = `
.c0 {
  -webkit-transition: -webkit-transform 0.2s;
  -webkit-transition: transform 0.2s;
  transition: transform 0.2s;
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

<svg
  aria-label="Arrow pointing up"
  class="c0"
  direction="up"
  height="12px"
  role="img"
  viewBox="0 0 11 5"
  width="12px"
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    arrow pointing 
    up
  </title>
  <g
    fill="none"
    fill-rule="evenodd"
    id="Symbols"
    stroke="none"
    stroke-width="1"
  >
    <g
      fill="black"
      fill-rule="nonzero"
      id="accordion/open"
      transform="translate(0.000000, -14.000000)"
    >
      <g
        id="Group"
      >
        <path
          d="M7.63149316,17.3813356 L4.28255792,21.7152282 C3.98915505,22.0949239 3.51345502,22.0949239 3.22005215,21.7152282 C3.07915512,21.5328919 3,21.2855908 3,21.0277282 L3,11.9722718 C3,11.4353009 3.33637072,11 3.75130504,11 C3.95056352,11 4.1416609,11.1024355 4.28255792,11.2847718 L7.63149316,15.6186645 C7.86830411,15.925124 7.92952055,16.0681625 7.96979452,16.2401176 C8.01006849,16.4120727 8.01006849,16.5879273 7.96979452,16.7598824 C7.92952055,16.9318375 7.86830411,17.074876 7.63149316,17.3813356 Z"
          id="path-1"
          transform="translate(5.500000, 16.500000) rotate(90.000000) translate(-5.500000, -16.500000) "
        />
      </g>
    </g>
  </g>
</svg>
`;
